File: Composicion/Ejercicio-3/Avion.py
class Parte:
    def __init__(self, nombre, peso):
        self.nombre = nombre
        self.peso = peso

    def get_nombre(self):
        return self.nombre

    def set_nombre(self, nombre):
        self.nombre = nombre

    def get_peso(self):
        return self.peso

    def set_peso(self, peso):
        self.peso = peso

    def mostrar_info(self):
        print(f"Nombre: {self.nombre} Peso: {self.peso}")


class Avion:
    def __init__(self, modelo, fabricante):
        self.modelo = modelo
        self.fabricante = fabricante
        self.partes = []

    def get_modelo(self):
        return self.modelo

    def set_modelo(self, modelo):
        self.modelo = modelo

    def get_fabricante(self):
        return self.fabricante

    def set_fabricante(self, fabricante):
        self.fabricante = fabricante

    def get_partes(self):
        return self.partes

    def set_partes(self, partes):
        self.partes = partes

    def agregar_parte(self, parte):
        self.partes.append(parte)

    def mostrar_avion(self):
        print("------------------------AVIÓN---------------------")
        print(f"Modelo: {self.modelo} Fabricante: {self.fabricante}")
        print("Partes del avión:")
        for parte in self.partes:
            parte.mostrar_info()


# MAIN
if __name__ == "__main__":
    a1 = Avion("Boeing 747", "Boeing")

    p1 = Parte("Motor", 3500)
    p2 = Parte("Alas", 8000)
    p3 = Parte("Tren de Aterrizaje", 1200)
    p4 = Parte("Cabina", 1500)

    a1.agregar_parte(p1)
    a1.agregar_parte(p2)
    a1.agregar_parte(p3)
    a1.agregar_parte(p4)

    a1.mostrar_avion()

File: Composicion/Ejercicio-3/Avion.java
import java.util.*;

class Avión {
    private String modelo, fabricante;
    private List<Parte> partes;
    
    public Avión(String m, String f)
    {
        this.modelo=m;
        this.fabricante=f;
        this.partes=new ArrayList<>();
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public String getFabricante() {
        return fabricante;
    }

    public void setFabricante(String fabricante) {
        this.fabricante = fabricante;
    }

    public List<Parte> getPartes() {
        return partes;
    }

    public void setPartes(List<Parte> partes) {
        this.partes = partes;
    }
    
    public void agregarParte(Parte p)
    {
        partes.add(p);
    }
    
    public void mostrarAvión()
    {
        System.out.println("------------------------AVION---------------------");
        System.out.println("Modelo: "+modelo+" "+"Fabricante: "+fabricante);
        System.out.println("Partes del avion: ");
        for(Parte p: partes)
        {
            p.mostrarInfo();
        }
    }
}

class Parte {
    private String nombre;
    private double peso;
    
    public Parte(String n, double p)
    {
        this.nombre=n;
        this.peso=p;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getPeso() {
        return peso;
    }

    public void setPeso(double peso) {
        this.peso = peso;
    }
    
    public void mostrarInfo()
    {
        System.out.println("Nombre: "+nombre+" "+"Peso: "+peso);
    }
}

class App {
    public static void main(String[] args) {
        
        Avión a1=new Avión("Boeing 747", "Boeing");
        
        Parte p1=new Parte("Motor", 3500);        
        Parte p2=new Parte("Alas", 8000);
        Parte p3=new Parte("Tren de Aterrizaje", 1200);
        Parte p4=new Parte("Cabina", 1500);
        
        a1.agregarParte(p1);        
        a1.agregarParte(p2);
        a1.agregarParte(p3);
        a1.agregarParte(p4);
        
        a1.mostrarAvión();
    }
}